### PR TITLE
Add fast checkout hydrateOrder endpoint

### PR DESCRIPTION
**Pull request by nicolenorman**
Add a new class which hits the fast checkout hydrate order endpoint. 

This will hydrate a Bold simple order type with data from a Magento quote.

### CODE REVIEW

File: Model/Quote/GetCartLineItems.php [hunk 1]
```diff
@@ -107,6 +108,7 @@ private function getLineItem(CartItemInterface $item): array
             'requires_shipping' => !$item->getProduct()->getIsVirtual(),
             'line_item_key' => (string)$item->getId(),
             'price' => $this->getLineItemPrice($item),
+            'vendor' => '',
```
Comment (nicolenorman) [inline Model/Quote/GetCartLineItems.php:111]: Note: Out of the box Magento doesn't have a vendor field for products so setting this to an empty string.

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,2 @@
+.idea
+.DS_Store
```
Comment (cxfos) [inline .gitignore:2]: 🙇 Thansk a lot.

**Review by cxfos — APPROVED** [review]

**Review by cxfos — APPROVED** [review]

File: Model/Order/HydrateOrderFromQuote.php [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+     * @param string $publicOrderId
+     * @return ResultInterface
+     */
+    public function hydrate(CartInterface $quote, string $publicOrderId): ResultInterface
+    {
+        $websiteId = (int)$quote->getStore()->getWebsiteId();
+        $billingAddress = $this->quoteToOrderAddressConverter->convert($quote->getBillingAddress());
+
+        if ($quote->getIsVirtual()) {
+            $totals = $quote->getBillingAddress()->getTotals();
```
Comment (mwraich) [inline Model/Order/HydrateOrderFromQuote.php:77]: 👏🏽 Ah, I forgot this piece. 
[Clarification] So Magento doesn't handle the difference between this? Trying to understand why the totals would be different?

File: Model/Order/HydrateOrderFromQuote.php [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+            'totals' => [
+                'sub_total' => $this->convertToCents($totals['subtotal']['value']),
+                'tax_total' => $this->convertToCents($totals['tax']['value']),
+                'discount_total' => $discountTotal,
+                'shipping_total' => $this->convertToCents($totals['shipping']['value']),
+                'order_total' => $this->convertToCents($totals['grand_total']['value'])
+            ],
+        ];
+
+        if ($quote->getCustomer()->getId()) {
```
Comment (mwraich) [inline Model/Order/HydrateOrderFromQuote.php:107]: [Clarification] This is for logged in users, correct?

File: Model/Order/HydrateOrderFromQuote.php [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+        return $taxLines;
+    }
+
+    /**
+     * Looks at total segments and makes unrecognized segments into fees and discounts
+     *
+     * @param array $totals
+     * @return array
+     */
+    private function getFeesAndDiscounts(array $totals): array
```
Comment (mwraich) [inline Model/Order/HydrateOrderFromQuote.php:164]: [Praise] 👏🏽 Thank you for taking the initiative on all of these edge cases

**Review by mwraich — COMMENTED** [review]
So looks good, testing on my shop though I got the following error: 

```
{
    "status": 422,
    "errors": [
        "{\"errors\":[\"Error at '\\\/totals\\\/discount_total': Value is not nullable\"],\"message\":\"Request validation failed for the reasons noted\"}"
    ],
    "body": []
}
```
I didn't have a discount present.

**Comment by nicolenorman** [comment]
> So looks good, testing on my shop though I got the following error:
> 
> ```
> {
>     "status": 422,
>     "errors": [
>         "{\"errors\":[\"Error at '\\\/totals\\\/discount_total': Value is not nullable\"],\"message\":\"Request validation failed for the reasons noted\"}"
>     ],
>     "body": []
> }
> ```
> 
> I didn't have a discount present.

Thank you for catching this @mwraich ! Just pushed a small change to account for no discounts.

**Review by mwraich — APPROVED** [review]

**Pull request merged**